quiche/src/recovery/congestion/resume.rs

```diff
@@ -45,18 +49,61 @@ impl std::fmt::Debug for Resume {
 
 impl Resume {
     pub fn new(trace_id: &str) -> Self {
+
+        // enabled will become false if either of the required CR ENV VARS is not supplied
+        let mut enabled = true;
+        let mut use_sr = true;
+
+        let mut previous_rtt = Duration::ZERO;
+        let mut previous_cwnd = 0;
+
+        if let Some(jw_oss) = std::env::var_os("PREVIOUS_CWND_BYTES")
+        {
+            if let Ok(jw_string) = jw_oss.into_string() {
+                if let Ok(jw_int) = jw_string.parse::<usize>() {
+                    previous_cwnd = jw_int;
+                }
+            }
+        } else {
+            enabled = false;
+        }
+
+        if let Some(rtt_oss) = std::env::var_os("PREVIOUS_RTT") 
+        {
+            if let Ok(rtt_string) = rtt_oss.into_string() {
+                if let Ok(rtt_int) = rtt_string.parse::<usize>() {
+                    previous_rtt = Duration::from_millis(rtt_int.try_into().unwrap());
+                }
+            }
+        } else {
+            enabled = false;
+        }
+
+        if let Some(no_sr_oss) = std::env::var_os("DISABLE_SR")
+        {
+            if let Ok(no_sr_string) = no_sr_oss.into_string() {
+                if let Ok(no_sr_int) = no_sr_string.parse::<usize>() {
+                    if no_sr_int == 1 {
+                        println!("Disabling SR");
+                        use_sr = false;
+                    }
+                }
+            }
+        }
+
```

Comment:
Reading from environment variables shouldn't be the default, there should be a config flag passed to enable this.

Reply:
Can you elaborate on the config flag? Do you mean, for example, a CLI argument passed to the server? I believe we are currently trying to limit these, and likely the final version would not be using environment variables at all.

Reply:
I mean that it should be a `bool` flag in `quiche::Config`.

Reply:
Adding that flag is easy. Accessing the flag when calling Resume fn new() will require to pass that in quiche::Config, as well as in RecoveryConfig so it can be accessed and then altering the calls to new(). Is this your preferred approach or do you have something neater in mind?